### PR TITLE
Update to php 7.4 type hinting

### PR DESCRIPTION
**Pull request by CarlSchwan**
Using rector, this also adds code style checks and some more ci pipelines

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

### CODE REVIEW

File: .github/workflows/lint-php.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      - main
+      - master
+      - stable*
+
+jobs:
+  php-lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        php-versions: ["7.4", "8.0"]
```
Comment (nickvergessen) [inline .github/workflows/lint-php.yml:21]:
```suggestion
        php-versions: ["7.4", "8.0", "8.1"]
```

File: psalm.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		<directory name="vendor" />
+		<ignoreFiles>
+			<directory name="vendor/phpunit/php-code-coverage" />
+		</ignoreFiles>
+	</extraFiles>
+	<issueHandlers>
+		<UndefinedClass>
+			<errorLevel type="suppress">
+				<referencedClass name="OC" />
+				<referencedClass name="Doctrine\DBAL\Types\Types" />
```
Comment (nickvergessen) [inline psalm.xml:28]: This should not be used anymore, there is OCP nowadays

File: rector.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<?php
```
Comment (nickvergessen) [inline rector.php:1]: not ment to be commited I think

File: psalm.xml (hunk 2 above)
Comment (CarlSchwan) [reply psalm.xml]: right and I ported everything to OCP in a previous PR :)

**Review by ArtificialOwl — APPROVED** [review]

**Pull request merged**